rel/expr_tuple.go

```diff
@@ -159,7 +159,8 @@ func (e *TupleExpr) Eval(local Scope) (Value, error) {
 			return nil, err
 		}
 	}
-	return tuple, nil
+	// TODO: Construct new tuple directly
+	return tuple.(*GenericTuple).Canonical(), nil
```

Comment:
As far as I can tell, the only utility of `Canonical` is to potentially convert the tuple into a list-item/string-char/etc tuple.

Reply:
That's exactly what it's for. Ideally, such forms should be detected upfront during tuple construction, which is happening already (`Canonical()` cheats by simply invoking it.) but sometimes you have an existing tuple that becomes a canonicalizable form through some transform and this method makes it easy to accommodate these scenarios.

Eventually, we need to tune this so that all tuple operations automatically return the canonical type and this function becomes redundant. But we're not there yet.